chapters/functions.tex

```diff
@@ -1775,27 +1775,30 @@ \subsubsection{Arrays}\label{arrays-1}
 
 Arrays are by default stored in row-major order when calling C functions and in column-major order when calling FORTRAN~77 functions.  These defaults can be overridden by the \lstinline!arrayLayout! annotation.  See the example below.
 
-The table below shows the mapping of an array argument in the absence of
-an explicit external function call when calling a C function. The type \lstinline!T!
-is allowed to be any of the simple types which can be passed to C as
+The table below shows the mapping of an array argument in the absence of an explicit external function call when calling a C function. 
+The type \lstinline!T! is allowed to be any of the simple types which can be passed to C as
 defined in \cref{simple-types} or a record type as defined in
 \cref{records} and it is mapped to the type \lstinline!T'! as defined in these sections
 for input arguments.
+Note that array inputs to C-function should be const-pointers to indicate that they should not be changed.
```

Comment:
As written, I'm sure this sentence is needed, but if so, I think it should say _C functions_ or _a C function_ (possibly with a non-breaking space).

Also aren't we trying to avoid _should not_ to leave less room for interpretation (as a tool generating code that calls this function, I'd rely on it actually not being changed)?

Making both of these change, we get a sentence that actually has some normative value (that it, it would be an error for the external function to cast away the constness and write to the memory):
```suggestion
Array inputs to a C function are const-pointers, indicating that they shall not be changed.
```

Reply:
I tried a different variant of this; I prefer functions here - and also clarifying that it's the **array** that cannot be changed.

Reply:
Looks good.  Resolving.